File: iossdkdemo/SDKDemo/APIControllers/GetScriptViewController.m
//
//  getScriptViewController.m
//  SDKDemo
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "getScriptViewController.h"
#import <OPGSDK/OPGSDK.h>
@interface GetScriptViewController ()

@end

@implementation GetScriptViewController

@synthesize surveyId;
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.surveyId.delegate=self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBAction
-(IBAction)getScript:(id)sender
{
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    //OPGScript *obj= [sdk getScript:self.surveyId.text error:&error];
//    if ([obj.isSuccess intValue]==1)
//    {
//        UIAlertController * alert = [UIAlertController
//                                     alertControllerWithTitle:@"OPGSDKv2.0"
//                                     message:@"Script Download Successful!"
//                                     preferredStyle:UIAlertControllerStyleAlert];
//        
//        UIAlertAction* okButton = [UIAlertAction
//                                   actionWithTitle:@"OK"
//                                   style:UIAlertActionStyleDefault
//                                   handler:^(UIAlertAction * action) {
//                                       //Handle your yes please button action here
//                                   }];
//        
//        [alert addAction:okButton];
//        [self presentViewController:alert animated:YES completion:nil];
//    }
//    else
//    {
//        UIAlertController * alert = [UIAlertController
//                                     alertControllerWithTitle:@"OPGSDKv2.0"
//                                     message:obj.statusMessage
//                                     preferredStyle:UIAlertControllerStyleAlert];
//        
//        UIAlertAction* okButton = [UIAlertAction
//                                   actionWithTitle:@"OK"
//                                   style:UIAlertActionStyleDefault
//                                   handler:^(UIAlertAction * action) {
//                                       //Handle your yes please button action here
//                                   }];
//        
//        [alert addAction:okButton];
//        [self presentViewController:alert animated:YES completion:nil];
//    }
}

#pragma mark - TextField delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField{
    
    [textField resignFirstResponder];
    return YES;
}
@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/GetScriptViewController.swift
//
//  GetScriptViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class GetScriptViewController: UIViewController, UITextFieldDelegate
{

    @IBOutlet weak var surveyId: UITextField!
    var obj : OPGScript?
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        surveyId.delegate=self
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        return true
    }
    
    
    @IBAction func getScript(_ sender: AnyObject)
    {
        let sdk = OPGSDK()        // Creating OPGSDK instance
        
        do {
           // obj = try sdk.getScript(self.surveyId.text)
        }
        catch{
            print("GetSCript Failed")         /* @"Error Occured. Contact Support!" */
            
        }
        
//        if ( obj?.isSuccess.int32Value==1)
//        {
//            let alertController = UIAlertController(title: "OPGSDKv2.0", message: "Script Download Successful!", preferredStyle: .alert)
//            
//            let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
//            alertController.addAction(defaultAction)
//            
//            present(alertController, animated: true, completion: nil)
//        }
//        else
//        {
//            let alertController = UIAlertController(title: "OPGSDKv2.0", message: "Script Download Failed!", preferredStyle: .alert)
//            
//            let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
//            alertController.addAction(defaultAction)
//            
//            present(alertController, animated: true, completion: nil)
//        }

    }

}
